modules/network-api/src/main/java/org/apache/ignite/network/serialization/MessageSerializationRegistry.java

```diff
@@ -25,22 +27,29 @@
  */
 public class MessageSerializationRegistry {
     /** message type -> MessageSerializerProvider instance */
-    private final MessageSerializationFactory<?>[] factories = new MessageSerializationFactory<?>[Short.MAX_VALUE << 1];
+    private final Map<Integer, MessageSerializationFactory<?>> factories = new HashMap<>();
```

Comment:
I disagree, this particular structure must be effective.
MessageSerializationFactory<?>[][] would make more sense.

Reply:
It makes more sense in terms of read performance, but does it make sense in terms of occupied memory? What is the expected number of message groups? Are we ok to have an array size of 2^16 * 2^16 elements? Or should we instead restrict the maximum number of group types and enforce that all these types should be in order?